shared/src/metabase/shared/parameters/parameters.cljc

```diff
@@ -4,51 +4,65 @@
       (:clj
        [(:require [clojure.string :as str]
                   [metabase.mbql.normalize :as mbql.normalize]
-                  [metabase.shared.util.i18n :refer [trs]])]
+                  [metabase.shared.util.i18n :refer [trs]]
+                  [metabase.util.date-2 :as u.date]
+                  [metabase.util.date-2.parse.builder :as b]
+                  [metabase.util.i18n.impl :as i18n.impl])
+        (:import java.time.format.DateTimeFormatter)]
        :cljs
        [(:require ["moment" :as moment]
                   [clojure.string :as str]
                   [metabase.mbql.normalize :as mbql.normalize]
                   [metabase.shared.util.i18n :refer [trs]])]))
 
-(defn- formatted-list
-  [values]
-  (str (str/join ", " (butlast values)) " " (trs "and") " " (last values)))
+#?(:cljs (comment moment/keep-me))
 
 (defmulti formatted-value
   "Formats a value appropriately for inclusion in a text card, based on its type. Does not do any escaping.
   For datetime parameters, the logic here should mirror the logic (as best as possible) in
   frontend/src/metabase/parameters/utils/date-formatting.ts"
-  (fn [tyype _value] (keyword tyype)))
+  (fn [tyype _locale _value] (keyword tyype)))
 
 (defmethod formatted-value :date/single
-  [_ value]
-  #?(:cljs (.format (moment value) "MMMM D, YYYY")
-     :clj value))
+  [_ locale value]
+  #?(:cljs (let [m (.locale (moment value) locale)]
+             (.format m "MMMM D, YYYY"))
+     :clj  (u.date/format "MMMM d, yyyy" (u.date/parse value) locale)))
 
 (defmethod formatted-value :date/month-year
-  [_ value]
-  #?(:cljs (let [m (moment value "YYYY-MM")]
+  [_ locale value]
+  #?(:cljs (let [m (.locale (moment value "YYYY-MM") locale)]
              (if (.isValid m) (.format m "MMMM, YYYY") ""))
-     :clj value))
+     :clj  (u.date/format "MMMM, yyyy" (u.date/parse value) locale)))
+
+#?(:clj
+   (def ^:private quarter-formatter-in
+     (b/formatter
+      "Q" (b/value :iso/quarter-of-year 1) "-" (b/value :year 4))))
+
+#?(:clj
+   (def ^:private quarter-formatter-out
+     (b/formatter
+      "Q" (b/value :iso/quarter-of-year 1) ", " (b/value :year 4))))
```

Comment:
These custom formatters might seem like overkill for just changing `-` to `,` but I figured there might be some locales that format quarters differently, so we'd want to do more than just simple string substitution here? Not sure.

Reply:
Definitely yes, use the real localization libraries for everything. There's always going to be some weird edge case.